docs/modules/publisher-directmethods.md

```diff
@@ -0,0 +1,328 @@
+OPC Publisher v2.8.2 implements the following [IoT Hub Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) which can be called from an application (anywhere in the world) leveraging the [IoT Hub Device SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-sdks).
+
+There are some direct methods which are inherited from 2.5.x in addition to new ones.
+
+Inherited direct methods from 2.5.x:
+
+- PublishNodes_V1
+- UnpublishNodes_V1
+- UnpublishAllNodes_V1
+- GetConfiguredEndpoints_V1
+- GetConfiguredNodesOnEndpoints_V1
+- GetDiagnosticInfo_V1
+
+New direct methods:
+
+-  AddOrUpdateEndpoints_V1
+
+Note: `_V1` suffix is not required.
+
+To understand the terminologies and migration path from version 2.5.x of OPC Publisher to 2.8.2, please refer to [this](publisher-dmmigration.md) migration document.
+
+
+Now let's dive into each direct method request and response payloads with examples.
+
+## PublishNodes_V1
+
+PublishNodes enables a client to add a set of nodes to be published for a specific [`DataSetWriter`](publisher-dmmigration.md#terminologies). When a `DataSetWriter` already exists, the nodes are incrementally added to the very same [`dataset`](publisher-dmmigration.md#terminologies). When it does not already exist, a new `DataSetWriter` is created with the initial set of nodes contained in the request.
+
+  *Request:* follows strictly the request [payload schema](publisher-dmmigration.md#payload-schema), the OpcNodes attribute being mandatory.
+
+  *Response:* when successful Status 200 and an empty json (`{}`) as payload
+
+  *Exceptions:* an exception is thrown when method call returns status other than 200
+
+  *Example:*
+
+  > *Method Name:* `PublishNodes_V1`
+  >
+  > *Request:*
+  >
+  > ```json
+  > {
+  >   "EndpointUrl": "opc.tcp://opcplc:50000/",
+  >   "DataSetWriterGroup": "Asset0",
+  >   "DataSetWriterId": "DataFlow0",
+  >   "DataSetPublishingInterval": 5000,
+  >   "OpcNodes":
+  >   [
+  >     {
+  >         "Id": "nsu=http://microsoft.com/Opc/OpcPlc/;s=FastUInt0",
+  >     }
+  >   ]
+  > }
+  > ```
+  >
+  > *Response:*
+  >
+  > ```json
+  > {
+  >   "status": 200,
+  >   "payload": {}
+  > }
+  > ```
+
+## UnpublishNodes_V1
+
+UnpublishNodes method enables a client to remove nodes from a previously configured DataSetWriter.
+
+*Note*: If all the nodes from a DataSet are to be unpublished, the DataSetWriter entity is completely removed from the configuration storage.
+
+  *Request:*  follows strictly the request payload schema, the OpcNodes attribute being mandatory.
+
+  *Response:* when successful - Status 200 and an empty json (`{}`) as Payload
+
+  *Exceptions:* a response corresponding to an exception will be returned if:
+  - request payload contains an endpoint (DataSet) that is not present in publisher configuration
+  - request payload contains a node that is not present in publisher configuration
+
+  *Example:*
+
+  > *Method Name:* `UnpublishNodes_V1`
```

Comment:
There should be an empty line between `Method Name:` and `Request:`, otherwise request is put on the same line when rendered. Like this:

```
  > *Method Name:* `UnpublishNodes_V1`
  >
  > *Request:*
```

There are several occurrences like this in this file. Please fix all of them.

Reply:
I was using Typora so many things were already rendered properly for me.
Added an extra line.